lxd/db/networks.go

```diff
@@ -299,32 +302,26 @@ func (c *ClusterTx) CreatePendingNetwork(node string, projectName string, name s
 		netType NetworkType
 	}{}
 
-	var errConsistency error
-	dest := func(i int) []any {
-		// Ensure that there is at most one network with the given name.
-		if i != 0 {
-			errConsistency = fmt.Errorf("More than one network exists with the given name")
-		}
-
-		return []any{&network.id, &network.state, &network.netType}
-	}
-
 	stmt, err := c.tx.Prepare("SELECT id, state, type FROM networks WHERE project_id = (SELECT id FROM projects WHERE name = ?) AND name=?")
 	if err != nil {
 		return err
 	}
 
 	defer func() { _ = stmt.Close() }()
 
-	err = query.SelectObjects(stmt, dest, projectName, name)
+	count := 0
+	err = query.SelectObjects(stmt, func(scan func(dest ...any) error) error {
+		// Ensure that there is at most one network with the given name.
+		if count != 0 {
+			return fmt.Errorf("More than one network exists with the given name")
```

Comment:
Can you use `return api.StatusErrorf(http.StatusConflict, "More than one network exists with the given name")` here?

Reply:
also `count` is never incremented, so this can never be reached?